src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectSnapshotChangeTrigger.cs

```diff
@@ -5,44 +5,50 @@
 using System.Collections.Generic;
 using System.Composition;
 using System.Diagnostics;
-using System.IO;
+using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
 
 namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
 {
+    [Shared]
     [Export(typeof(ProjectSnapshotChangeTrigger))]
     internal class WorkspaceProjectSnapshotChangeTrigger : ProjectSnapshotChangeTrigger
     {
+        private readonly ProjectWorkspaceStateGenerator _workspaceStateGenerator;
```

Comment:
Do we need another thing for this? Tag Helper discovery is already a service.

Reply:
The WorkspaceStateGenerator utilizes the TagHelperService. I created a ProjectWorkspaceStateGenerator with the intent that we'd have more logic in there at some point (more information derived from the Roslyn project)

Reply:
This is the code review feedback I gave you before I looked at the PR. This seems too coupled, this should all be one class.